### ISSUE TITLE
"Copy page title" option on the sidebar history context menu

### ISSUE THREAD

**Issue by ericpa06** [aminomancer/uc.css.js#33]
Hi, would it be possible to add a **"Copy page title"** option when dealing with the sidebar menu history panel? Like, so that in the same way you could copy the page URL you could also copy the page title. All the extension I tried so far didn't enable an option in that particular context menu. Something like this:
![image_3023](https://user-images.githubusercontent.com/6937720/156880016-84208be0-252b-483b-ab4d-0860825f5fbb.png)
.

**Comment by aminomancer** [comment]
So, when you hit the "Copy" menu item, it doesn't simply copy the page URL, it copies the [places node](https://searchfox.org/mozilla-central/source/browser/components/places/PlacesUIUtils.jsm#1638-1679) (a bookmark or history entry), which means [serializing its data](https://searchfox.org/mozilla-central/source/toolkit/components/places/PlacesUtils.jsm#1000-1120) so it can be added to the clipboard. The actual clipboard item winds up having multiple representations and an app will just use the first compatible one, which in most cases will be the URL. But this is what allows you to copy/cut the bookmark and paste it into another bookmark folder. So that you're not just copying the URL, you're effectively copying the actual bookmark, and how that gets pasted depends on the application and context. So, if adding a new Copy menu item, you'd have to decide whether you want it to work the same way, or just really simply copy the title.

If you just want it to copy the title, then really an extension is the best approach, because there's a bunch of extension circuitry for adding context menu items to browsers. It's not easy for an autoconfig script to add context menu items to the sidebar. I have a few scripts that do that and they're really big. Because every time you open a sidebar it creates a new browser with a new window and new context menu. For the scripts I made, it makes more sense because the context menu items do things that extensions simply can't do. But copying a places node's title is trivial. I'm really not sure why an extension that purports to do this wouldn't have that feature. Can you link the extension(s) you're testing?

**Comment by ericpa06** [comment]
Yeah, I also thought that this should be achievable through extensions as well, especially because there are extensions that add such options on other context menus, like when you click on a tab and you are able to copy its title, like this
![image](https://user-images.githubusercontent.com/6937720/156904899-3e0b2202-40df-4f6a-b4ea-1e40b88e569e.png)

Currently I'm using this one here _(which adds the options to copy the page title on a tab by right clicking on it)_
https://addons.mozilla.org/firefox/addon/copytabtitleurl/

Although I test other as well, and none of them added that option on the sidebar history context menu. Maybe there is already some extension buried somewhere on firefox addons site that does that, if I understood correctly what you said, this option is technically possible to be implemented through a extensions.

**Comment by benzBrake** [comment]
Agree with an extension is the best approach. A relavetive extension is bookmark tab here(BTH), BTH adds a menuitem to sidebar context menu. May be you can modify BTH to add this funciton yourself.

This morning I have add this function to my bookmark operate script and I found that is easy to make it work with history popup. You can try it: https://github.com/benzBrake/FirefoxCustomize/blob/master/userChromeJS/BookmarkOpt.uc.js
This script do not support multi-selection.

![copytitle](https://user-images.githubusercontent.com/11323774/169742484-fa9eb965-ffd5-4fc3-b723-9f90bbb42020.png)

**Comment by ericpa06** [comment]
Thank you so much [benzBrake](https://github.com/benzBrake), it worked flawlessly.
